src/__swaps__/screens/Swap/hooks/useSearchCurrencyLists.ts

```diff
@@ -23,156 +22,161 @@ export interface AssetToBuySection {
+    (data: TokenSearchResult) => {
+      const matchingNetwork = data.filter(asset => asset.chainId === state.toChainId);
+      return matchingNetwork.slice(0, 20).filter(asset => asset.icon_url);
+    },
+    [state.toChainId]
   );
-  const isCrosschainSearch = useMemo(() => {
-    return inputChainId && inputChainId !== toChainId;
-  }, [inputChainId, toChainId]);
 
-  // provided during swap to filter token search by available routes
-  const fromChainId = useMemo(() => {
-    return isCrosschainSearch ? inputChainId : undefined;
-  }, [inputChainId, isCrosschainSearch]);
-
-  const queryIsAddress = useMemo(() => isAddress(query), [query]);
-  const keys: TokenSearchAssetKey[] = useMemo(() => (queryIsAddress ? ['address'] : ['name', 'symbol']), [queryIsAddress]);
-  const threshold: TokenSearchThreshold = useMemo(() => (queryIsAddress ? 'CASE_SENSITIVE_EQUAL' : 'CONTAINS'), [queryIsAddress]);
-
-  // static search data
   const { data: verifiedAssets, isLoading: verifiedAssetsLoading } = useTokenSearch(
     { list: 'verifiedAssets' },
-    { select: filterNonTokenIconAssets, staleTime: 60 * 60 * 1000, cacheTime: 24 * 60 * 60 * 1000 }
+    {
+      select: query.length > 0 ? filterNonTokenIconAssets : sliceTopResultsByNetworkWithCoinIconUrls,
+      staleTime: 60 * 60 * 1000,
+      cacheTime: 24 * 60 * 60 * 10000,
+    }
   );
 
-  // current search
+  const memoizedData = useMemo(() => {
+    const fromChainId = state.isCrosschainSearch ? state.fromChainId : undefined;
+    const queryIsAddress = isAddress(query);
+    const keys: TokenSearchAssetKey[] = queryIsAddress ? ['address'] : ['name', 'symbol'];
+    const threshold: TokenSearchThreshold = queryIsAddress ? 'CASE_SENSITIVE_EQUAL' : 'CONTAINS';
+    const enableUnverifiedSearch = query.length > 2;
+
+    const bridgeAsset = state.isCrosschainSearch
+      ? verifiedAssets?.find(asset => isLowerCaseMatch(asset.mainnetAddress, state.assetToSellAddress))
+      : null;
+    const filteredBridgeAsset = bridgeAsset && filterBridgeAsset({ asset: bridgeAsset, filter: query }) ? bridgeAsset : null;
+
+    return {
+      isCrosschainSearch: state.isCrosschainSearch,
+      fromChainId,
+      queryIsAddress,
+      keys,
+      threshold,
+      enableUnverifiedSearch,
+      verifiedAssetsForChain: verifiedAssets,
+      filteredBridgeAsset,
+    };
+  }, [state.assetToSellAddress, state.fromChainId, state.isCrosschainSearch, query, verifiedAssets]);
+
+  const getAddressForChainId = useCallback((chainId: ChainId, token: RainbowToken) => {
+    if (chainId === ChainId.mainnet) {
+      const mainnetAddress = token.networks[chainId].address;
+      return mainnetAddress ?? token.address;
+    }
+    return token.networks[chainId].address;
+  }, []);
+
+  const { favoritesMetadata: rawFavorites } = useFavorites();
+  const favorites = useMemo(() => Object.values(rawFavorites), [rawFavorites]);
+
+  const unfilteredFavorites = useMemo(() => {
+    return Object.values(favorites)
```

Comment:
you're calling Object.values twice here. Also think memoizing favorites is unnecessary since it's only being used in unfilteredFavorites which is already memoized.